### PR TITLE
Add 'interval' provider

### PR DESCRIPTION
**Pull request by namhyung**
Hello!

The interval provider is for something needs to be done periodically.  I think  the most common usecase is to print maps and optionally clear them.  So I added it as well like below:

```
# cat readhist.ply
kretprobe:vfs_read {
    @r["bytes"] = quantize(retval);
}
interval:s:1 {
    print(@r);
    clear(@r);
}

# ply -c 'sleep 3' readhist.ply
...
```

The syntax of the interval is same as bpftrace and users need to specify unit and time.  To handle maps (and aggregations) in the print and clear function, I added a new MAPREF (map reference) type.  The functions will rewrite a map argument to a map reference to pass a map fd.

### CODE REVIEW

File: src/libply/aux/perf_event.c [hunk 1]
```diff
@@ -73,6 +73,42 @@ int perf_event_attach(struct ply_probe *pb, const char *path,
...
+	attr.disabled = 1;
+
+	fd = perf_event_open(&attr, pid, cpu, group_fd, 0);
+	if (fd < 0)
+		return -errno;
+
+	/* if (ioctl(fd, PERF_EVENT_IOC_ENABLE, 0)) { */
+	/* 	close(fd); */
+	/* 	return -errno; */
+	/* } */
```
Comment (chenhengqi) [inline src/libply/aux/perf_event.c:99]: debug leftover?

File: src/libply/provider/interval.c [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+		if (unit[1] != 's') {
+			_e("invalid time unit: %s\n", unit);
+			return -1;
+		}
+		/* nano-seconds */
+		break;
+	default:
+		_e("unsupported time unit: %s\n", unit);
+		return -1;
+	}
```
Comment (chenhengqi) [inline src/libply/provider/interval.c:80]: Something like `interval:ss:1` or `interval:nss:100` are accepted by this check, right ?

File: src/libply/aux/perf_event.c (hunk 1 above)
Comment (namhyung) [reply src/libply/aux/perf_event.c:99]: Right, it's just copied from the perf_event_attach function, but we might need to remove it as it's enabled later.

File: src/libply/provider/interval.c (hunk 2 above)
Comment (namhyung) [reply src/libply/provider/interval.c:80]: Thanks for your review, will change.

**Comment by namhyung** [comment]
Forgot to leave a comment.  I've updated it to get rid of the first commit and to change the syntax as you requested.  Now it's default to second so simply write as `interval:1` (or even `i:1`) for 1 second interval.  It also accepts `interval:1s` and `interval:500ms` and so on.  Please take a look.

**Comment by wkz** [comment]
Awesome, thanks a lot!

**Pull request merged**